bionic/core/flow_execution.py

```diff
@@ -162,12 +164,15 @@ def _process_active_entry(self, entry):
             self._aip_execution_enabled
             and entry.state.func_attrs.aip_task_config is not None
         )
-        can_compute_remotely = (
-            (aip_execution_enabled_for_entry or self._parallel_execution_enabled)
-            and entry.state.should_persist
-            and not entry.state.task.is_simple_lookup
-        )
-        if can_compute_remotely:
+        entry_can_be_computed_remotely = (
+            aip_execution_enabled_for_entry or self._parallel_execution_enabled
+        ) and entry.state.should_persist
+        if entry_can_be_computed_remotely:
+            remote_subgraph = RemoteSubgraph(entry.state, self._bootstrap)
+            if not remote_subgraph.all_states_can_be_serialized:
+                entry_can_be_computed_remotely = False
+
+        if entry_can_be_computed_remotely:
```

Comment:
Did you mean to remove this? The block above also has `entry_can_be_computed_remotely`

Reply:
I think this is what I intended; maybe it's confusing that the block above modifies `entry_can_be_computed_remotely`? The idea is that we need to meet all the initial conditions for remote computation, and then we need to also check that the subgraph is serializable. Maybe instead of modifying `entry_can_be_computed_remotely`, it would be better to introduce a new variable?

Reply:
I think adding a new variable would be cleaner than modifying the existing one.